src/Service/Request.service.js

```diff
@@ -41,18 +41,47 @@ export default class RequestService extends DependencyAwareClass {
     return typeof queryParameters[parameter] !== 'undefined' ? queryParameters[parameter] : ifNull;
   }
 
+  /**
+   * Get all HTTP headers included in the request.
+   *
+   * @returns {object} An object with a key for each header.
+   */
+  getAllHeaders() {
+    return { ...this.getContainer().getEvent().headers };
+  }
+
+  /**
+   * Get an HTTP header from the request.
+   *
+   * The header name is case-insensitive.
+   *
+   * @param {string} name The name of the header.
+   * @param {string} [whenMissing] Value to return if the header is missing.
+   *   (default: empty string)
+   *
+   * @returns {string}
+   */
+  getHeader(name: string, whenMissing: string = '') {
```

Comment:
What are your thoughts on moving this method into `DependencyInjection` where the event is stored? Then here we can still define a shortcut, but the this header utility would be available to all `DependencyAware` objects

Reply:
Yeah, I was in two minds about this. `DependencyInjection` does contain the event and I've been using `di.getEvent().headers` in some other places, but my mental model is that headers are part of the request, so the `RequestService` seems the right place to put them. I think the same is true of search parameters and POST bodies, which are part of the event but exposed via the `RequestService`.

Reply:
I agree with you and that's why I am very happy for these methods to be specified to `RequestService`. Just wandering whether they should be accessible to other dependencies.

But now that I think about it, other services can surely request `getContainer.get(DEFINITIONS.REQUEST`), right?

Reply:
Approved 🙂

Reply:
Yeah, exactly, they can easily get the request via `di`/`getContainer()`.